ui/src/App.vue

```diff
@@ -2,45 +2,31 @@
   <div class="ui container">
     <h2 class="ui center aligned header">Job Hooroom</h2>
       <SearchBar />
-      <div class="flex-box">
-        <GraphCard :locations="locations"></GraphCard>
-        <GraphCard :employmentTypes="employmentTypes"></GraphCard>
-        <GraphCard :skills="skills"></GraphCard>
-        <GraphCard :isRemote="isRemote"></GraphCard>
+      <div v-if="this.graphData" class="flex-box">
+        <GraphCard v-for="(value, index) in this.graphData.data" :key="index" :graphProp="value.buckets" :title="index" />
       </div>
   </div>
 </template>
 
 <script>
-import axios from 'axios';
-
 import SearchBar from './components/SearchBar';
 import GraphCard from './components/GraphCard';
 
-const mockAPI = process.env.VUE_APP_MOCK_API;
 
 export default {
   name: 'App',
   components: {
     SearchBar,
     GraphCard
   },
-  data() {
+  data: function() {
     return {
-      isRemote: [],
-      employmentTypes: [],
-      locations: [],
-      skills: []
+      graphData: null
     }
   },
-  mounted() {
-    axios.get(mockAPI)
-        .then(response => {
-          this.isRemote = response.data.is_remote.buckets;
-          this.employmentTypes = response.data.employment_type.buckets;
-          this.locations = response.data.location.buckets;
-          this.skills = response.data.skills.buckets;
-        })
+  async beforeCreate() {
+    await this.$store.dispatch('setAnalytics');
+    this.graphData = this.$store.state.graph;
```

Comment:
Not a too big deal, but can we use Vuex getters instead of accessing the state directly?

Reply:
Added getters to set value for this.graphData